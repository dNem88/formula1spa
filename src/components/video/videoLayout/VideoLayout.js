import React, {useState,useEffect} from 'react'
import styles from './videoLayout.module.css'
import hVideos from '../../../utils/api/heroku_api/hVideos'
import ErrorComp from '../../common/errorComp/ErrorComp'
import VideosContainer from '../videosContainer/VideosContainer'
import {useLocation} from 'react-router-dom'

function VideoLayout() {
    const [state, setState] = useState({error: null, errorMessage: null, videos: null, mainID: null, mainVideo: null})
    const location = useLocation()
    let id = location.pathname.match(/\/[a-z0-9-]{10,}/g);
    
    useEffect(() => {
        async function FetchVideos() {
            try {
                const response = await hVideos.getVideos();
                if (!response.ok) {
                    throw new Error('Failed to fetch')
                }
                let json = await response.json()
                setState({error: null, errorMessage: null, videos: json, mainID: null})
            } catch(err) {
                setState({error: true, errorMessage: err.message, videos: null, ...state})
            }
           
        }
         FetchVideos()
    }, [])

    useEffect(() => {
        if (id && state.videos) {
            let videoToRender = state.videos.filter(x => x._id === id[0].substring(1))
            setState({...state, mainID: id[0].substring(1), mainVideo: videoToRender[0]})
        }
        if (!id) {
            if (state.videos) {
                id = state.videos[0]._id
                 setState({
                     ...state,
                     mainID: id,
                     mainVideo: state.videos[0]
                 })
            }
        }
    }, [state.videos, location])

    return (
        <div className={styles.layout}>
            {state.error && state.errorMessage ? <ErrorComp errorMessage={state.errorMessage}/> : null}
            <h1 className={styles['header']}>Video</h1>
            <VideosContainer {...state}/>
        </div>
        
    )
}

export default VideoLayout