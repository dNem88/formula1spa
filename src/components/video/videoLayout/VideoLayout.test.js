import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import VideoLayout from './VideoLayout'



jest.mock('../videosContainer/VideosContainer', () => () => 'VideosContainer')
jest.mock('../latestVideosContainer/LatestVideosContainer', () => () => 'LatestVideosContainer')


describe('Testing VideosLayout', () => {
    
    it('Render Properly', () => {
        const snapshot = renderer.create(
        <MemoryRouter>
            <VideoLayout/>
        </MemoryRouter>
    ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('div')
        expect(snapshot.props.className).toBe('layout')
        expect(snapshot.children.length).toBe(4)
        expect(snapshot.children[0].children[0]).toBe('Video')
        expect(snapshot.children[2].children[0]).toBe('Latest videos')
    })
})