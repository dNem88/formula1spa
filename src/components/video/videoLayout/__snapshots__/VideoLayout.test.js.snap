// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideosLayout Render Properly 1`] = `
<div
  className="layout"
>
  <h1
    className="header"
  >
    Video
  </h1>
  VideosContainer
  <h1
    className="latest-header"
  >
    Latest videos
  </h1>
  LatestVideosContainer
</div>
`;
